plans/TOOLING.md

```diff
@@ -0,0 +1,241 @@
+# THIS IS A WORK IN PROGRESS
+
+This document describes the tentative roadmap for AsyncAPI tooling in the next ~6 months and the strategies we'll follow to create the tools.
+
+## Roadmap
+We want to start developing tools as we make progress with version 2.0.0. To minimize the impact of future specification changes in the tools (_just those for v2 while it's in development_), we'll perform [all the breaking changes of the spec](https://github.com/asyncapi/asyncapi/issues?utf8=%E2%9C%93&q=label%3A%22v2+breaking+change%22+) first, and then we'll start with tooling **development**. Notice I've used the word _development_ because I want to make it clear that we can start other types of work beforehand, like planning, architecting, documenting the approach, etc. Actually, this document is already a pre-development work.
+
+### Milestones
+
+#### :arrow_right: Create a Parser/Validator
+The first tool we would have to implement is the AsyncAPI parser/validator. The parser/validator is a core piece of all the other tooling we'll build in the future. Therefore it's a top priority for us.
+
+##### Requirements:
+1. It MUST support YAML and JSON as input formats.
+2. It MUST support the following schema formats: OpenAPI schemas, JSON Schema Draft 04-07, Protobuf, and Avro.
+3. It MUST support plugging in a new format parser.
+4. It MUST be implemented as a Go package and [exported as a C shared object](https://medium.com/learning-the-go-programming-language/calling-go-functions-from-other-languages-4c7d8bcc69bf).
+5. It MUST have wrappers in the following languages: Java and Node.js.
+
+#### :arrow_right: Create code generation framework
```

Comment:
Agree yeoman is cool for CLI, but we should also offer non-CLI code generation. @rmelian pointed out to reusing some parts of Swagger Codegen. We can actually use both as they are totally different things.

Reply:
ok, so you mean to support also dynamic code generation, makes sense, I think it is common for node.js apps. I suggest to make it clear in the description

Reply:
I didn't go forward in the description because the requirements are still not clear. Only the parser requirements are already defined. The rest are there for "road mapping" purposes. We'll get into implementation details afterwards.